src/Avalonia.Visuals/Rendering/SceneGraph/RectangleNode.cs

```diff
@@ -105,8 +105,28 @@ public override void Render(IDrawingContextImpl context)
             }
         }
 
-        // TODO: This doesn't respect CornerRadius yet.
         /// <inheritdoc/>
-        public override bool HitTest(Point p) => Bounds.Contains(p);
+        public override bool HitTest(Point p)
+        {
+            // TODO: This doesn't respect CornerRadius yet.
```

Comment:
I think it might be best to add rounded rectangles to the scene graph as `GeometryNode`s to get this working.

Or we could do the maths ourselves, if anyone knows what that maths looks like.

Reply:
Hmm. We could do that but it wouldn't be optimal in terms of performance. The change would be minimal.

Reply:
Borders with rounded corners are currently missing a geometry for clipping. Maybe we could add that and use that also for hit testing. We can still use the fast render path for uniform borders.